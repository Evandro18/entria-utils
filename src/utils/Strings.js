// @flow
export const isEmpty = (value: string): boolean => value === undefined || value === null || value === '';

export const isAlpha = (value: string): boolean => /^[a-zA-ZÁÉÍÓÚáéíóúÃÕãõçÇ ]+$/.test(value);

export const isEmail = (value: string): boolean => /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value);

export const uppercaseFirstLetter = (text: string): string => `${text.charAt(0).toUpperCase()}${text.slice(1)}`;

export const capitalize = (text: string): string =>
  text.toLowerCase().split(' ').map(word => uppercaseFirstLetter(word)).join(' ');

export const numbers = (value: string): string => value.replace(/\D/g, '');
