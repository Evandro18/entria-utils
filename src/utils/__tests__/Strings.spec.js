import * as Strings from '../Strings';

it('isEmpty - should return true if null', () => {
  expect(Strings.isEmpty(null)).toMatchSnapshot();
});

it('isEmpty - should return true if undefined', () => {
  expect(Strings.isEmpty(undefined)).toMatchSnapshot();
});

it('isEmpty - should return true if empty string', () => {
  expect(Strings.isEmpty('')).toMatchSnapshot();
});

it('isEmpty - should return false if filled', () => {
  expect(Strings.isEmpty('awesome')).toMatchSnapshot();
});

it('isEmpty - should return false if 0', () => {
  expect(Strings.isEmpty('0')).toMatchSnapshot();
});

it('isAlpha - should return true if valid', () => {
  expect(Strings.isAlpha('Fábio Assunção')).toMatchSnapshot();
});

it('isAlpha - should return false if invalid', () => {
  expect(Strings.isAlpha('Fábio Assunção #@*&')).toMatchSnapshot();
});

it('isEmail - should return true if valid', () => {
  expect(Strings.isEmail('john@doe.com')).toMatchSnapshot();
});

it('isEmail - should return false if invalid', () => {
  expect(Strings.isEmail('fake@d.c.b')).toMatchSnapshot();
});

it('uppercaseFirstLetter', () => {
  expect(Strings.uppercaseFirstLetter('barney stinson')).toMatchSnapshot();
});

it('capitalize', () => {
  expect(Strings.capitalize('barney stinson')).toMatchSnapshot();
});

it('numbers', () => {
  expect(Strings.numbers('1, 2, 3, Un pasito p\' adelante. 1, 2, 3, Un pasito p\' atrás.')).toMatchSnapshot();
});
