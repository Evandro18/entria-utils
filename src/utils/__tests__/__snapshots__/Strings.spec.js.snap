// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`capitalize 1`] = `"Barney Stinson"`;

exports[`isAlpha - should return false if invalid 1`] = `false`;

exports[`isAlpha - should return true if valid 1`] = `true`;

exports[`isEmail - should return false if invalid 1`] = `false`;

exports[`isEmail - should return true if valid 1`] = `true`;

exports[`isEmpty - should return false if 0 1`] = `false`;

exports[`isEmpty - should return false if filled 1`] = `false`;

exports[`isEmpty - should return true if empty string 1`] = `true`;

exports[`isEmpty - should return true if null 1`] = `true`;

exports[`isEmpty - should return true if spaces string 1`] = `true`;

exports[`isEmpty - should return true if undefined 1`] = `true`;

exports[`numbers 1`] = `"123123"`;

exports[`uppercaseFirstLetter 1`] = `"Barney stinson"`;
